python/ray/experimental/serve/router/routers.py

```diff
@@ -79,21 +80,37 @@ def register_actor(
             init_args: List = [],
             init_kwargs: dict = {},
             num_replicas: int = 1,
-            max_batch_size: int = -1,  # Unbounded batch size
+            max_batch_size: int = -1 # Unbounded batch size
     ):
         """Register a new managed actor.
         """
         self.managed_actors[actor_name] = actor_class
         self.actor_init_arguments[actor_name] = (init_args, init_kwargs)
         self.max_batch_size[actor_name] = max_batch_size
-
-        ray.experimental.get_actor(self.name).set_replica.remote(
+        ray.experimental.get_actor(self.name).set_replication_factor.remote(
             actor_name, num_replicas)
 
-    def get_actor(self, actor_name):
-        return self.managed_actors[actor_name]
+    def change_all_actors(self, actor_name):
+
+        assert actor_name in self.managed_actors, (
+            ACTOR_NOT_REGISTERED_MSG(actor_name))
 
-    def set_replica(self, actor_name, new_replica_count):
+        current_replicas = len(self.actor_handles[actor_name])
+
+        for _ in range(current_replicas):
+            # Note actor destructor will be called after all remaining
+            # calls finish. Therefore it's safe to call del here.
+            del self.actor_handles[actor_name][-1]
+
+        # Increase the number of replicas
+        for _ in range(current_replicas):
+            args = self.actor_init_arguments[actor_name][0]
+            kwargs = self.actor_init_arguments[actor_name][1]
+            new_actor_handle = self.managed_actors[actor_name].remote(
```

Comment:
I don't think just passing `num_cpus` etc to the remote actor init method will work. For me, I have to change the private attribute in the actor class to make it work. (cc @robertnishihara is this the only way to dynamiclly pass in required resource)?

```py
        actor_class = self.managed_actors[actor_name]
        actor_class._num_cpus = num_cpus
        actor_class._resources = resource_vector
        new_actor_handle = actor_class.remote(
            *init_args, **init_kwargs
        )
```

Reply:
nvm we should call ```
    def _remote(self,
                args=None,
                kwargs=None,
                num_cpus=None,
                num_gpus=None,
                resources=None):
        """Create an actor.
```

Reply:
When I make this change I get one test failure. I'm not sure why yet but if it is obvious to you please let me know what I should change.

`def test_change_replication_factor(router: DeadlineAwareRouter):

    assert ray.get(router.get_replication_factor.remote("ManagerApiTestActor")) == 1

    router.set_replication_factor.remote("ManagerApiTestActor", 10)
    assert ray.get(router.get_replication_factor.remote("ManagerApiTestActor")) == 10`